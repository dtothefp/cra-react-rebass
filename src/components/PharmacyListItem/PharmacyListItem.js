import PropTypes from 'prop-types';
import {Box, Text} from 'rebass';
import React, { memo } from 'react';

const Item = ({children}) => <Text as="p" py={1}>{children}</Text>;

Item.propTypes = {
  children: PropTypes.node
};

const PharmacyListItem = memo(({
  address: {
    city, state, line1, zip_code: zip
  },
  name,
  phone,
}) => (
  <Box
    my={2}
    p={2}
    sx={{
      borderStyle: `solid`,
      borderWidth: `1px`,
      borderColor: `primary`,
    }}
  >
    <Item as="p">{name}</Item>
    <Item as="p">{line1}, {city}, {state} {zip}</Item>
    <Item as="p">{phone}</Item>
  </Box>
));

PharmacyListItem.propTypes = {
  address: PropTypes.shape({
    city: PropTypes.string,
    state: PropTypes.string,
    line1: PropTypes.string,
    zip_code: PropTypes.string,
  }),
  name: PropTypes.string,
  phone: PropTypes.string,
};

export default PharmacyListItem;
