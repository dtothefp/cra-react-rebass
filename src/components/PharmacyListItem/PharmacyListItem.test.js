import React from 'react';
import { render } from '@testing-library/react';
import PharmacyListItem from './PharmacyListItem';
import pharmacyMock from '../../__mocks__/pharmacy.item';

const {address, name, phone_number: phone} = pharmacyMock;

describe(`#App`, () => {
  it(`renders without crashing`, () => {
    const {container} = render(
      <PharmacyListItem
        address={address}
        name={name}
        phone={phone}
      />
    );

    expect(container).toMatchSnapshot();
  });
});

