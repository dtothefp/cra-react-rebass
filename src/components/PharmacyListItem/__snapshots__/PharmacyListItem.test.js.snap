// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#App renders without crashing 1`] = `
<div>
  <div
    class="css-xbdrcs"
  >
    <p
      class="css-143f1i2"
    >
      CVS Pharmacy
    </p>
    <p
      class="css-143f1i2"
    >
      255 Main St
      , 
      Venice
      , 
      CA
       
      90291
    </p>
    <p
      class="css-143f1i2"
    >
      (310) 399-2436
    </p>
  </div>
</div>
`;
